### PR TITLE
Add Pagination Support To /admin/user Endpoint

### PR DESCRIPTION
**Pull request by qur786**
Description:
• Implemented pagination support for the GET /api/v1/admin/user endpoint.
• Added skip, take query parameters to control pagination.
• Modified service and repository logic to handle new parameters.

Code Changes:
• Updated controller to accept new query parameters.
• Adjusted service method to handle pagination.

### CODE REVIEW

File: apps/api/src/app/admin/admin.service.ts [hunk 1]
```diff
@@ -429,8 +429,14 @@ export class AdminService {
...
-  public async getUsers(): Promise<AdminUsers> {
-    return { users: await this.getUsersWithAnalytics() };
+  public async getUsers({
+    skip,
+    take = Number.MAX_SAFE_INTEGER
+  }: {
+    skip?: number;
+    take?: number;
+  }): Promise<AdminUsers> {
+    return { users: await this.getUsersWithAnalytics({ take, skip }) };
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:439]: Can you swap `skip` and `take` to preserve the alphabetical order please?

File: apps/api/src/app/admin/admin.service.ts [hunk 2]
```diff
@@ -640,7 +646,13 @@ export class AdminService {
     return { marketData, count: marketData.length };
   }
 
-  private async getUsersWithAnalytics(): Promise<AdminUsers['users']> {
+  private async getUsersWithAnalytics({
+    take,
+    skip
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:651]: Can you swap `skip` and `take` to preserve the alphabetical order please?

File: apps/api/src/app/admin/admin.service.ts [hunk 3]
```diff
@@ -640,7 +646,13 @@ export class AdminService {
...
+  }: {
+    take?: number;
+    skip?: number;
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:654]: Can you swap `skip` and `take` to preserve the alphabetical order please?

File: apps/api/src/app/admin/admin.service.ts [hunk 4]
```diff
@@ -662,6 +674,8 @@ export class AdminService {
     const usersWithAnalytics = await this.prismaService.user.findMany({
       orderBy,
       where,
+      skip,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:677]: Please move up ⬆️

File: apps/api/src/app/admin/admin.service.ts [hunk 5]
```diff
@@ -662,6 +674,8 @@ export class AdminService {
...
+      take,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:678]: Please move up ⬆️

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for your contribution, @qur786! 😎
I have added a few comments keep the code base consistent.

File: apps/api/src/app/admin/admin.service.ts (hunk 1 above)
Comment (qur786) [reply apps/api/src/app/admin/admin.service.ts]: Done.

File: apps/api/src/app/admin/admin.service.ts (hunk 2 above)
Comment (qur786) [reply apps/api/src/app/admin/admin.service.ts]: Done.

File: apps/api/src/app/admin/admin.service.ts (hunk 3 above)
Comment (qur786) [reply apps/api/src/app/admin/admin.service.ts:653]: Done.

File: apps/api/src/app/admin/admin.service.ts (hunk 4 above)
Comment (qur786) [reply apps/api/src/app/admin/admin.service.ts:676]: Done.

File: apps/api/src/app/admin/admin.service.ts (hunk 5 above)
Comment (qur786) [reply apps/api/src/app/admin/admin.service.ts:677]: Done.

**Review by dtslvr — APPROVED** [review]
Thank you for your first contribution, @qur786! 🙌🏻
Ready to merge now.

**Comment by dtslvr** [comment]
@qur786 I have created a follow-up issue: https://github.com/ghostfolio/ghostfolio/issues/4053

**Pull request merged**